tcms/static/js/testrun_actions.js

```diff
@@ -757,11 +757,19 @@ function initialize_addlink_dialog() {
     },
     buttons: {
       "OK": function() {
-        var name = jQ('#testlog_name').attr('value');
-        var url = jQ('#testlog_url').attr('value');
-        var is_defect = $('#is_defect').is(':checked');
-        var update_tracker = $('#update_tracker').is(':checked');
-        var case_id = dialog_p.dialog('option', 'case_id');
+        var name = jQ('#testlog_name').attr('value'),
+            url = jQ('#testlog_url').attr('value'),
+            is_defect = $('#is_defect').is(':checked'),
+            update_tracker = $('#update_tracker').is(':checked'),
+            case_id = dialog_p.dialog('option', 'case_id'),
+            tmpAElement = document.createElement('a');
+
+        tmpAElement.href = url;
+
+        //check if url is valid
+        if (url.length === 0 || (tmpAElement.host && tmpAElement.host === window.location.host)) {
```

Comment:
Why comparing the host part of the URL to the host part of the current window? In case of using the built-in defect tracker both will be the same so your check will return. Not what we want I think.

Reply:
Hmm, I missed that. Basically if a.href = 'invalidurl' the a.host will be current host. I don't want to use a regex here. I will try to find different solution.

Reply:
Just search for `://` inside the value. This looks good enough of a URL indicator.